www/docs/server/merging-routers.md

```diff
@@ -66,6 +66,69 @@ export const userRouter = router({
 
 ```
 
+Under the hood, nested procedure paths are strings in dot notation (`'.'` as delimiter) that contain the procedure name preceeded by all namespaces in the path from the root router.
+
+```ts title='client.ts'
+const users = await client.user.list.query(); // => [];
+// underlying path is "user.list"
+```
+
+You can also configure a custom namespace delimiter:
+
+```ts twoslash title='server.ts'
+// @filename: trpc.ts
+import { initTRPC } from '@trpc/server';
+
+const t = initTRPC.create({
+  namespaceDelimiter: '/' as const,
+});
+
+export const middleware = t.middleware;
+export const router = t.router;
+export const publicProcedure = t.procedure;
+
+// @filename: routers/_app.ts
+import { router } from '../trpc';
+import { z } from 'zod';
+
+import { userRouter } from './user';
+
+const appRouter = router({
+  user: userRouter,
+});
+
+export type AppRouter = typeof appRouter;
+
+// @filename: routers/user.ts
+import { router, publicProcedure } from '../trpc';
+import { z } from 'zod';
+
+export const userRouter = router({
+  list: publicProcedure.query(() => {
+    return [];
+  }),
+});
+```
+
+```ts title='client.ts'
+import { createTRPCProxyClient, httpBatchLink } from '@trpc/client';
+import type { AppRouter } from '../../server/trpc';
+
+const client = createTRPCProxyClient<AppRouter>({
+  links: [
+    httpBatchLink({
+      url: 'http://localhost:3000/trpc',
+    }),
+  ],
+  namespaceDelimiter: '/', // this has to match the delimiter on the server
+});
+
+const users = await client.user.list.query(); // => [];
+// underlying path is "user/list"
+```
+
+**Note**: Choosing a custom namespace delimiter is a must when using AWS HTTP APIs since dots are not allowed in paths.
```

Comment:
I don't think this is true? I have hosted my tRPC backend on both API Gateway "REST" and "HTTP" to front a lambda (we had to move between for infra reasons) and didn't have any problems with dots.

If this statement is true there must be some more nuanced information which is being omitted, or the name "HTTP APIs" is confusing 🤔

Reply:
_right_ #3537 is some interesting nuance :) 

The thing I don't understand in that example is _why_ you would want to reproduce the routes into API Gateway, I just have a simple proxy set up and it works flawlessly. Best case doing this would break batching. 

Anyway I'm a bit off track, I wonder if what this topic really needs is an AWS Serverless hosting doc which covers the nuances and recommendations in depth? This statement as it's written has the chance to be quite misleading to readers :)

Reply:
Removed. :)

Reply:
> The thing I don't understand in that example is why you would want to reproduce the routes into API Gateway, I just have a simple proxy set up and it works flawlessly. Best case doing this would break batching.

I guess some might want more fine-grained control over memory size, provisioning, etc. and separate monitoring and logs by having lambdas dedicated to specific paths.